src/games/strategy/triplea/delegate/BattleDelegate.java

```diff
@@ -100,7 +113,16 @@ public void start() {
       addBombardmentSources();
       m_needToAddBombardmentSources = false;
     }
+    fightCurrentBattle();
+    if (rocketHelper != null && !games.strategy.triplea.Properties.getRollRocketsAfterSBR(getData())) {
+      rocketHelper.fireRockets(m_bridge, m_bridge.getPlayerID());
+      rocketHelper = null;
```

Comment:
Why is rocketHelper so stateful, why do we have to keep setting it to null all the time?
There is a big code smell with rocketHelper, dual usage. It being null has one meaning, otherwise it being not null means something else. This is a hidden context and tribal knowledge surrounding every usage of the variable, it makes this class even more complicated.